src/shared/copy.h

```diff
@@ -30,6 +30,14 @@ typedef enum CopyFlags {
         COPY_GRACEFUL_WARN = 1 << 15, /* Skip copying file types that aren't supported by the target filesystem */
 } CopyFlags;
 
+typedef enum DenyType {
+        DENY_DONT,
```

Comment:
i'd say `= 0` here explicitly, and add a comment `/* we want INT_TO_PTR(DENY_DONT) to map to NULL*/`

and then further down even:

```c
assert_cc(INT_TO_PTR(DENY_DONT) == NULL);
```